README.md

```diff
@@ -74,97 +79,92 @@ df.columns = ['x', 'y', 'z1']
-conda activate koalas-dev-env
-conda install -c conda-forge pyspark=2.4
-conda install -c conda-forge --yes --file requirements-dev.txt
-pip install -e .  # installs koalas from current checkout
-```
 
-Once setup, make sure you switch to `koalas-dev-env` before development:
-```bash
-conda activate koalas-dev-env
-```
+## Design Principles
 
-### Running Tests
+This section outlines design princples guiding the Koalas project.
 
-There is a script `./dev/pytest` which is exactly same as `pytest` but with some default settings to run Koalas tests easily.
+### Be Pythonic
 
-To run all the tests, similar to our CI pipeline:
-```bash
-# Run all unittest and doctest
-./dev/pytest
-```
+Koalas targets Python data scientists. We want to stick to the convention that users are already familiar with as much as possible. Here are some examples:
 
-To run a specific test file:
-```bash
-# Run unittest
-./dev/pytest -k test_dataframe.py
+- Function names and parameters use snake_case, rather than CamelCase. This is different from PySpark's design. For example, Koalas has `to_pandas()`, whereas PySpark has `toPandas()` for converting a DataFrame into a pandas DataFrame. In limited cases, to maintain compatibility with Spark, we also provide Spark's variant as an alias.
 
-# Run doctest
-./dev/pytest -k series.py --doctest-modules databricks
-```
+- Koalas respects to the largest extent the conventions of the Python numerical ecosystem, and allows the use of numpy types, etc. that are supported by Spark.
 
-To run a specific doctest/unittest:
-```bash
-# Run unittest
-./dev/pytest -k "DataFrameTest and test_Dataframe"
+- Koalas docs' style and infrastructure simply follow rest of the PyData projects'.
 
-# Run doctest
-./dev/pytest -k DataFrame.corr --doctest-modules databricks
-```
+### Unify small data (pandas) API and big data (Spark) API, but pandas first
 
-Note that `-k` is used for simplicity although it takes an expression. You can use `--verbose` to check what to filter. See `pytest --help` for more details.
+The Koalas DataFrame is meant to provide the best of pandas and Spark under a single API, with easy and clear conversions between each API when necessary. When Spark and pandas have similar APIs with subtle differences, the principle is to honor the contract of the pandas API first.
 
-### Building Documentation
+There are 4 different classes of functions:
 
-To build documentation via Sphinx:
+ 1. Functions that are only found in Spark (`select`, `selectExpr`). These functions should also be available in Koalas.
 
-```bash
-cd docs && make clean html
-```
+ 2. Functions that are found in Spark but that have a clear equivalent in pandas, e.g. `alias` and `rename`. These functions will be implemented as the alias of the pandas function, but should be marked that they are aliases of the same functions. They are provided so that existing users of PySpark can get the benefits of Koalas without having to adapt their code.
```

Comment:
I think `alias` and `rename` are not necessary. Existing users of PySpark will be confused what APIs are exposed and not. I was thinking it's unclear what Koalas API expects. This can be worked around via `to_spark`. We can expose another API that truncates Spark's index.

PySpark doesn't have index so when PySpark users use `to_spark(index='truncate').alias(...)` won't get confused.

Reply:
These functions are used everywhere in pyspark user code, so I would hate to have to rename everything to use koalas. Also, kdf.to_spark(index='truncate').alias(...).to_koalas() is also way too long to be useful.

Reply:
If this is added since it's commonly used, I am okie.

Reply:
I changed the wording to say 

```There are different classes of functions:

 1. Functions that are found in both Spark and pandas under the same name (`count`, `dtypes`, `head`). The return value is the same as the return type in pandas (and not Spark's).
    
 2. Functions that are found in Spark but that have a clear equivalent in pandas, e.g. `alias` and `rename`. These functions will be implemented as the alias of the pandas function, but should be marked that they are aliases of the same functions. They are provided so that existing users of PySpark can get the benefits of Koalas without having to adapt their code.
 
 3. Functions that are only found in pandas. When these functions are appropriate for distributed datasets, they should become available in Koalas.
 
 4. Functions that are only found in Spark that are essential to controlling the distributed nature of the computations, e.g. `cache`. These functions should be available in Koalas.

We are still debating whether data transformation functions only available in Spark should be added to Koalas, e.g. `select`. We would love to hear your feedback on that.```